### ISSUE TITLE
Filling a form doesn't generate events

### ISSUE THREAD

**Issue by toupeira** [passff/passff#182]
I ran into an issue with Grafana, when I fill the login form with PassFF the login button stays disabled because it's apparently listening for key events ([similar issue in older Grafana version here](https://github.com/grafana/grafana/issues/3055)). I didn't encounter this issue with LastPass so I decided to open a bug report here, instead of reopening the Grafana issue :)

If I trigger a `change` event on the input fields (using `$('input').change()` in the browser console) the login button becomes active, maybe you could add this in PassFF after a form fill?

**Comment by tuxor1337** [comment]
Nice idea, but we don't have jQuery and using plain JavaScript (`.dispatchEvent(new Event("change"))`, same with "keyup" and "keydown"), I don't get the expected result since Grafana is listening to other events. Any other suggestions?

**Comment by grodzik** [comment]
How about something like this?:

```
diff --git a/src/modules/page.js b/src/modules/page.js
index 6ee488c..9eda4a7 100644
--- a/src/modules/page.js
+++ b/src/modules/page.js
@@ -55,20 +55,37 @@ PassFF.Page = {
   },

   setInputs: function(doc, passwordData) {
     PassFF.Page.setLoginInputs(doc, passwordData.login);
     PassFF.Page.setPasswordInputs(doc, passwordData.password);
     PassFF.Page.setOtherInputs(doc, passwordData._other);
   },

+  fireOnChange: function(doc, element) {
+    var evt = doc.createEvent("HTMLEvents");
+    evt.initEvent("change", false, true);
+    element.dispatchEvent(evt);
   },

   setLoginInputs: function(doc, login) {
     PassFF.Page.getLoginInputs(doc).forEach(function(loginInput) {
       loginInput.value = login;
+      PassFF.Page.fireOnChange(doc, loginInput);
     });
   },

   setPasswordInputs: function(doc, password) {
     PassFF.Page.getPasswordInputs(doc).forEach(function(passwordInput) {
       passwordInput.value = password;
+      PassFF.Page.fireOnChange(doc, passwordInput);
     });
   },

@@ -82,6 +99,7 @@ PassFF.Page = {
       }
       if (value) {
         otherInput.value = value;
+        PassFF.Page.fireOnChange(doc, otherInput);
       }
     });
   },
```

I messed a bit with it some time ago, but since it was my first attempt in touching Firefox plugins, I had hard times making good use of Firefox Developer. This code worked for me for example on Grafana site, but made some sites that used to work to stop working - due to lack of time and experience didn't manged to get it fully working, but maybe it will be useful.

**Comment by tuxor1337** [comment]
This is a patch for the legacy version of PassFF, right? I don't have time to adapt it to and test it with the latest version of PassFF, but it's certainly a promising start, if somebody has the time to look into this.

**Comment by gutierri** [comment]
@toupeira Is there somewhere I can simulate this?

**Comment by davidar** [comment]
https://www.hover.com/signin does this too, but even manually dispatching a change event doesn't seem to help...

**Comment by cstroe** [comment]
I've hit this trying to log in to capitalone.com:

![capitalone-login-form](https://user-images.githubusercontent.com/1182093/27983941-0f76cc68-638f-11e7-95ff-5cffd3b81f44.png)

A (tedious) workaround is to edit the password field and type a character then backspace.

**Comment by tuxor1337** [comment]
A fix similar to the proposal of @toupeira has been implemented in PR #232 by GordanV. But this does not fix the problem for https://www.hover.com/signin ... Any ideas on that?

**Comment by tuxor1337** [comment]
Solved, all credits go to @GordanV and @drmoose :+1:
